Autopilot/PathManager/Src/pathStateClasses.cpp

```diff
@@ -120,4 +164,287 @@ pathManagerState& fatalFailureMode::getInstance()
+        //retrieving desired altitude for slope state and setting it 
+        cruisingState::_outputdata.desiredAltitude = LandingManager::changingAltitude(sensorFusion::sensorInput,landingTransitionStage::path.aimingPoint, landingTransitionStage::path.intersectionPoint, landingTransitionStage::path.stoppingPoint);
+        //retrieving desired speed for approach speed and setting it
+        cruisingState::_outputdata.desiredSpeed = LandingManager::approachSpeed(getFromTelemetry::telemetryInput.windSpeed, getFromTelemetry::telemetryInput.packageOnBoard);
+    }
+
+    if(landingTransitionStage::waypointStatus == INVALID_PARAMETERS)
+    {
+        pathMgr -> isError = true;
+    }
+    
+    if(pathMgr->isError)
+    {
+        pathMgr -> setState(fatalFailureMode::getInstance());
+    }
+    else
+    {
+        pathMgr -> setState(coordinateTurnElevation::getInstance());
+    }
+}
+
+pathManagerState& landingSlopeStage::getInstance()
+{
+    static landingSlopeStage singleton;
+    return singleton;
+}
+
+void landingFlareStage::execute(pathManager* pathMgr)
+{
+    if(sensorFusion::sensorInput.altitude <= (DECRAB_ALTITUDE + getFromTelemetry::telemetryInput.stoppingAltitude)) //altitude is below 70 cm
+    {
+        pathMgr->stage = DECRAB;
+    }
+    else
+    {
+        //maintaining horizontal position
+        landingTransitionStage::waypointStatus = landingTransitionStage::landingPath.get_next_directions(sensorFusion::sensorInput, &cruisingState::_outputdata);
+        
+        //throttle off
+        cruisingState::_outputdata.controlDetails.throttlePassby = true;
+        cruisingState::_outputdata.controlDetails.throttlePercent = 0;
+
+        //maintaing speed for flare attitude
+        cruisingState::_outputdata.desiredSpeed = LandingManager::slowFlightSpeed(getFromTelemetry::telemetryInput.packageOnBoard);
+    }   
+
+    if(landingTransitionStage::waypointStatus == INVALID_PARAMETERS)
+    {
+        pathMgr -> isError = true;
+    }
+    
+
+    if(pathMgr->isError)
+    {
+        pathMgr -> setState(fatalFailureMode::getInstance());
+    }
+    else
+    {
+        pathMgr -> setState(coordinateTurnElevation::getInstance());
+    }
+}
+
+pathManagerState& landingFlareStage::getInstance()
+{
+    static landingFlareStage singleton;
+    return singleton;
+}
+
+void landingDecrabStage::execute(pathManager* pathMgr)
+{
+    if(sensorFusion::sensorInput.altitude <= (TOUCHDOWN_ALTITUDE + getFromTelemetry::telemetryInput.stoppingAltitude)) //altitude is 5 cm or less/ultrasonic sensor sensed 5cm or less
+    {
+        pathMgr->stage = TOUCHDOWN;
+    }
+    else
+    {
+        //align heading with landing direction
+        cruisingState::_outputdata.desiredHeading = getFromTelemetry::telemetryInput.stoppingDirectionHeading;
+        //retrieving desired slow flight speed
+        cruisingState::_outputdata.desiredSpeed = LandingManager::slowFlightSpeed(getFromTelemetry::telemetryInput.packageOnBoard);
```

Comment:
Why are you accessing crusingState parameters?

Reply:
Since coordinated turns would need to be pulling instructions from a wide variety of states, I was thinking to modify one variable for all the control states. Now that I think about it, I think it's better for all states to modify an input variable in coordinate turns instead of cruising. What do you think?